crt_portal/static/js/complaint_view_filters.js

```diff
@@ -267,13 +318,49 @@
     }
 
     props.el.addEventListener('change', function(event) {
-      filterDataModel[props.name] = event.target.value;
-      if (props.name == 'per_page') {
+      const grouping = document.getElementsByName('grouping')[0].value;
+      if (props.name == 'per_page' && grouping !== 'default') {
+        const per_page_els = document.getElementsByName('per_page');
+        filterDataModel['group_params'] = updateGroupParams(
+          filterDataModel['group_params'],
+          per_page_els
+        );
+      } else {
+        filterDataModel[props.name] = event.target.value;
+      }
+      if (props.name == 'per_page' || props.name == 'grouping') {
         dom.getElementById('apply-filters-button').click();
       }
     });
   }
 
+  function updateGroupParams(group_params, per_page_els) {
+    if (group_params.length) {
+      group_params = JSON.parse(group_params[0]?.replaceAll('"', "'").replaceAll("'", '"'));
+      for (let i = 0; i < per_page_els.length; i++) {
+        group_params[i]['per_page'] = per_page_els[i].value ? Number(per_page_els[i].value) : 15;
+      }
+    } else {
+      for (let i = 0; i < per_page_els.length; i++) {
+        group_params.push({
+          page: 1,
+          per_page: per_page_els[i].value ? Number(per_page_els[i].value) : 15,
+          sort: []
+        });
+      }
+    }
+    return JSON.stringify(group_params);
+  }
```

Comment:
two readability techniques applied here:
- Reduce nesting using guard clauses
- Prefer Array functions / loops that don't surface the index where possible.

```suggestion
  function updateGroupParams(group_params, per_page_els) {
    if (group_params.length) {
      group_params = JSON.parse(group_params[0]?.replaceAll('"', "'").replaceAll("'", '"'));
      per_page_els.forEach((el, i) => {
        group_params[i]['per_page'] = el.value ? Number(el.value) : 15;
      });
      return JSON.stringify(group_params);
    }

    return JSON.stringify(per_page_els.map(el => {
      return {
        page: 1,
        per_page: el.value ? Number(el.value) : 15,
        sort: []
      }
    }));
  }
```